### PR TITLE
Fix validation of memory resource as it should not be configured in millibytes

### PR DESCRIPTION
**Pull request by aveshagarwal**
@kubernetes/rh-cluster-infra @derekwaynecarr

### CODE REVIEW

**Comment by aveshagarwal** [comment]
This validation can be done using golang regexp too, let me know if that is desired than the current implementation in this PR, and I will update.

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -2754,10 +2759,22 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
...
+			allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), "memory request must not be configured in millibytes"))
+		}
 	}
 	return allErrs
 }
 
+// validateResourceRequirementsMemory ensures that memory is not configured in milli value
+func IsResourceRequirementsMemoryValid(quantity resource.Quantity) bool {
+	memStr := quantity.String()
+	if memStr[len(memStr)-1] == 'm' {
```
Comment (lavalamp) [inline pkg/api/validation/validation.go]: This is not the right way to test it. You need to check that rounding to whole units doesn't change the value.

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -2754,10 +2759,22 @@ func ValidateResourceRequirements(requirements *api.ResourceRequirements, fldPat
...
 		}
+		if resourceName == api.ResourceMemory && !IsResourceRequirementsMemoryValid(quantity) {
+			allErrs = append(allErrs, field.Invalid(fldPath, quantity.String(), "memory request must not be configured in millibytes"))
+		}
 	}
 	return allErrs
 }
 
+// validateResourceRequirementsMemory ensures that memory is not configured in milli value
+func IsResourceRequirementsMemoryValid(quantity resource.Quantity) bool {
```
Comment (derekwaynecarr) [inline pkg/api/validation/validation.go]: There should be a test case that demonstrates the failure scenario that is covered by this change.

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (derekwaynecarr) [inline pkg/api/validation/validation.go]: agreed... i had a similar need when working on https://github.com/openshift/origin/pull/8409/files#diff-5fd1547857aea1ef0cdb18cb293f88faR162  but punted on at it at the time.  you can see the same rounding logic there though.

**Comment by lavalamp** [comment]
This is aging, so I'll close it. I do want this change though, so please send another PR with the comment fixed when it's convenient for you!

**Pull request closed without merging**